experimenter/queue.py

```diff
@@ -25,24 +26,79 @@ def is_running():
 
 def get_queue(queue_name: str = _DEFAULT_QUEUE) -> rq.Queue:
     return rq.Queue(queue_name, connection=get_connection())
+    
+    
+def get_worker_message(workers: list, queue):
+    num_workers = len(workers)
+    message = 'number of workers on queue {}: {}'.format(queue.name, num_workers)
+    for it, worker in enumerate(workers):
+        success = worker.successful_job_count
+        fail = worker.failed_job_count
+        message = message+'\n\t\t\t worker: {}'.format(it)
+        message = message+'\n\t\t\t\t number of successful jobs: {}'.format(success)
+        message = message+'\n\t\t\t\t number of failed jobs: {}'.format(fail) 
+    return message
+    
+
+def get_failed_job(queue_name:str = _DEFAULT_QUEUE, job_num:int = 0):
+    #pass name and connection
+    reg = rq.registry.FailedJobRegistry(name = queue_name, connection = get_connection())
+    job_ids = reg.get_job_ids()
+    if (len(job_ids)<=0):
+        return "None", reg
+    job_id = job_ids[0]
+    return job_id, reg
+    
+    
+def save_failed_job(queue_name:str = _DEFAULT_QUEUE, job_num:int = 0):
+    if (queue_name is None):
+        queue_name = _DEFAULT_QUEUE
+    job_id, failed_queue = get_failed_job()
+    job = rq.job.Job.fetch(job_id, connection=get_connection())
+    with open (os.path.join('/data',"failed_job_{}.txt".format(job_num)), 'w') as job_file:
+        job_file.write(job.exc_info)
+    #remove the job
+    failed_queue.remove(job_id, delete_job=True)
+    print(_SAVE_FAILED_MESSAGE.format(os.path.join('/data',
+                              "failed_job_{}.txt".format(job_num))))
```

Comment:
These two functions look like they have not been run. Did you use past versions of it? How was it helpful?

Reply:
The first function, I use for returning the failed queue so that I can empty it, and also I return a single job. The returning of the single job is used in the second function, which then saves the failure output to a file in the '/data' directory. This is for remote development and debugging. We can look at the reason why a task failed without having to do some kind of dashboard thing (it will be saved to a file). I use that one with the queue --save-failed command. The first function is used in quite a few places on the queue.py file. It returns the failed queue, which is what I use it for most often.